bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -228,16 +228,259 @@ def __set__(self, instance, value):
         instance._udf_resources = tuple(value)
 
 
-def _build_udf_resources(resources):
+class AbstractQueryParameter(object):
+    """Base class for named / positional query parameters.
     """
-    :type resources: sequence of :class:`UDFResource`
-    :param resources: fields to be appended.
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct paramter from JSON resource.
 
-    :rtype: mapping
-    :returns: a mapping describing userDefinedFunctionResources for the query.
+        :type resource: dict
+        :param resource: JSON mapping of parameter
+
+        :rtype: :class:`ScalarQueryParameter`
+        """
+        raise NotImplementedError
+
+    def to_api_repr(self):
+        """Construct JSON API representation for the parameter.
+
+        :rtype: dict
+        """
+        raise NotImplementedError
+
+
+class ScalarQueryParameter(AbstractQueryParameter):
+    """Named / positional query parameters for scalar values.
+
+    :type name: str or None
+    :param name: Parameter name, used via `@foo` syntax.  If None, the
```

Comment:
Double backticks are probably what you want?

Throughout the changed/added files.